app/jobs/rest_of_world_zones/update_job.rb

```diff
@@ -0,0 +1,58 @@
+module RestOfWorldZones
+  class UpdateJob < ApplicationJob
+    def perform(spree_member, rest_of_world_country_zone, profile_id)
+      @spree_member = spree_member
+      @rest_of_world_country_zone = rest_of_world_country_zone
+      @profile_id = profile_id
+      update_rest_of_world_zones
+    end
+
+    def update_rest_of_world_zones
+      @spree_member.is_a?(Spree::Country) ? update_rest_of_world_country_zone : update_rest_of_world_state_zone
+    end
+
+    def update_rest_of_world_country_zone
```

Comment:
but what are you updating here?
I just see that you're destroying specific record

Reply:
I update zone by destroying its member

Reply:
ok, got it